public/app/plugins/panel/timeseries/plugins/OutsideRangePlugin.tsx

```diff
@@ -25,37 +26,41 @@ export const OutsideRangePlugin = ({ config, onChangeTimeRange }: ThresholdContr
     });
   }, [config]);
 
+  // If we don't have enough time values or if we can't
+  // change the time range then we don't display zoom to data
   if (timevalues.length < 2 || !onChangeTimeRange) {
     return null;
   }
 
+  // If we don't have an appropriate time range then we don't know if data is
+  // being displayed
   if (!timeRange || !timeRange.time || !timeRange.min || !timeRange.max!) {
     return null;
   }
 
   // Time values are always sorted for uPlot to work
-  let i = 0,
-    j = timevalues.length - 1;
+  let i = 0;
+  let j = timevalues.length - 1;
 
-  while (i <= j && timevalues[i] == null) {
-    i++;
-  }
+  // Increment i until we get the last non-null value
+  for (; i < timevalues.length && timevalues[i] == null; i++) {}
 
-  while (j >= 0 && timevalues[j] == null) {
-    j--;
-  }
+  // Decrement j until we get the first non-null-value
+  for (; j >= 0 && timevalues[j] == null; j--) {}
```

Comment:
we probably don't need to scan all the way back to 0, but just `i`

```suggestion
  for (; j >= i && timevalues[j] == null; j--) {}
```

Reply:
also, can j end up -1 here due to `>=`? then will try to access `timevalues[-1]`?